paimon-core/src/main/java/org/apache/paimon/append/AppendOnlyWriter.java

```diff
@@ -279,7 +279,11 @@ public long memoryOccupancy() {
 
     @Override
     public void flushMemory() throws Exception {
-        flush(false, false);
+        if (sinkWriter.bufferSpillableWriter()) {
```

Comment:
just same to `MergeTreeWriter`?

Reply:
`MergeTreeWriter` seems don't have this bug.